File: frontend/src/model/Expense.js
export class Expense {
    constructor(
        id = undefined,
        category = '',
        amount = 0,
        description = '',
        date = new Date().toISOString().split('T')[0]
    ) {
        this.id = id;
        this.category = category;
        this.amount = amount;
        this.description = description;
        this.date = date;
    }
}

File: services/expense-service/model/expense.go
package model

import (
	"time"

	"github.com/google/uuid"
)

// Expense represents a single transaction (income or expense) record.
type Expense struct {
	ID          uuid.UUID `gorm:"type:uuid;primaryKey" json:"id"`
	UserID      string    `gorm:"index" json:"userId"`
	Type        string    `json:"type"` // "INCOME" or "EXPENSE"
	Amount      float64   `json:"amount"`
	Category    string    `json:"category"`
	Description string    `json:"description,omitempty"`
	Date        time.Time `json:"date"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}
